pincer/objects/guild/guild.py

```diff
@@ -482,6 +484,222 @@ async def modify_member(self, _id: int, **kwargs) -> GuildMember:
+        topic: Optional[str] = None,
+        bitrate: Optional[int] = None,
+        user_limit: Optional[int] = None,
+        rate_limit_per_user: Optional[int] = None,
+        position: Optional[int] = None,
+        permission_overwrites: Optional[List[Overwrite]] = None,
+        parent_id: Optional[Snowflake] = None,
+        nsfw: Optional[bool] = None
+    ) -> Channel:
+        """|coro|
+        Create a new channel object for the guild.
+
+        Parameters
+        ----------
+        name : str
+            channel name (1-100 characters)
+        type : Optional[:class:int`]
+            the type of channel
+        topic : Optional[:class:str`]
+            channel topic (0-1024 characters)
+        bitrate : Optional[:class:`int`]
+            the bitrate (in bits) of the voice channel (voice only)
+        user_limit : Optional[:class:`int`]
+            the user limit of the voice channel (voice only)
+        rate_limit_per_user : Optional[:class:`int`]
+            amount of seconds a user has to wait before sending another message (0-21600)
+            bots, as well as users with the permission manage_messages or manage_channel, are unaffected
+        position : Optional[:class:`int`]
+            sorting position of the channel
+        permission_overwrites : Optional[List[:class:`~pincer.objects.guild.overwrite.Overwrite`]]
+            the channel's permission overwrites
+        parent_id : Optional[:class:`~pincer.utils.snowflake.Snowflake`]
+            id of the parent category for a channel
+        nsfw : Optional[:class:`bool`]
+            whether the channel is nsfw
+
+        """
+        ...
+
+    async def create_channel(self, **kwargs):
+        data = await self._http.post(f"guilds/{self.id}/channels", data=kwargs)
+        return Channel.from_dict(construct_client_dict(self._client, data=data))
+
+    async def modify_channel_positions(
+        self, *channel: Dict[str, Optional[Union[int, bool, Snowflake]]]
+    ):
+        """|coro|
+        Create a new channel object for the guild.
+
+        Parameters
+        ----------
+
+        \\*channel : Dict[str, Optional[Union[int, bool, :class:`~pincer.utils.snowflake.Snowflake`]
+            Keys:
+                - id : :class:`~pincer.utils.snowflake.Snowflake`
+                - position : Optional[:class:`int`]
+                - lock_permissions : Optional[:class:`bool`]
+                - parent_id : Optional[:class:`~pincer.utils.snowflake.Snowflake`]
+
+        """
+
+    async def list_active_threads(self) -> Tuple[
+        List[Union[PublicThread, PrivateThread]], List[GuildMember]]:
+        """|coro|
+        Returns all active threads in the guild, including public and private threads.
+        """
+        data = await self._http.get(f"guilds/{self.id}/threads/active")
+
+        threads = [Channel.from_dict(channel) for channel in data["threads"]]
+        members = [GuildMember.from_dict(member) for member in data["members"]]
```

Comment:
Make these generators

Reply:
data["thread"] and data["member] aren't the same length so I don't think there's a way to do it

Reply:
Making them generator expressions should be enough, so in that case it's really just replacing `[...]` with `(...)`

Reply:
yeah That Would Be Good